src/audio/pipeline.c

```diff
@@ -14,6 +14,7 @@
 #include <sof/drivers/timer.h>
 #include <sof/lib/alloc.h>
 #include <sof/lib/mailbox.h>
+#include <sof/lib/mm_heap.h>
 #include <sof/lib/uuid.h>
```

Comment:
My understanding is that:
- the API is exposed in alloc.h
- inner details are handled in mm_heap.h.

So, mm_heap.h should only be included by alloc.h and the rest of the users should only include alloc.h

Reply:
Declarations in mm_heap.h are not just "inner details". Imo it is completely different set of APIs (heap management on the kernel level mostly) for different parts of the FW. Actually the separation should go a step further and move the implementation of heap mgmt API closer to the infrastructure/kernel related things while alloc.h/.c is a part of the library used by clients to access memory allocation API. All those clients do not kneed to know structures like mm, mm_heap etc. nor functions like init_heap(), free_heap() etc. All this kernel level heap mgmt should be accessible to application level client code. So I'd not include them indirectly.

The code referred above is a piece that uses both APIs, there is `heap_trace_all()` call in the code, thus both includes are there.

Reply:
I see. ok.